src/coreclr/tools/aot/ObjWriter/llvm.patch

```diff
@@ -1,3 +1,15 @@
+diff --git a/include/llvm/BinaryFormat/COFF.h b/include/llvm/BinaryFormat/COFF.h
+index b395db6eaa8..74820a8f79f 100644
+--- a/include/llvm/BinaryFormat/COFF.h
++++ b/include/llvm/BinaryFormat/COFF.h
+@@ -395,6 +395,7 @@ enum RelocationTypesARM64 {
+   IMAGE_REL_ARM64_ADDR64 = 0x000E,
+   IMAGE_REL_ARM64_BRANCH19 = 0x000F,
+   IMAGE_REL_ARM64_BRANCH14 = 0x0010,
++  IMAGE_REL_ARM64_REL32 = 0x0011,
```

Comment:
Have you looked at how hard it would be to update to newer LLVM that includes this change?

Reply:
The latest [Arch64WinCOFFObjectWriter.cpp](https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/AArch64/MCTargetDesc/AArch64WinCOFFObjectWriter.cpp) in LLVM still does not generate `IMAGE_REL_ARM64_REL32` relocations, so I would have to change that file anyway.

The version of LLVM we currently use takes 0.78 GiB for the repository + 1.6 GiB for build artifacts.  The latest LLVM takes 2.5 GiB for  the repository + 2.3 GiB for build artifacts. It takes significantly longer to clone and build. Most importantly, incremental build of LLVM with MSBuild and `-DLLVM_OPTIMIZED_TABLEGEN=1` is broken and rebuilds almost everything (35 minutes on my machine). Need to follow up on that.